Course/Course/Presentation/Container/CourseContainerViewModel.swift

```diff
@@ -140,6 +143,9 @@ public class CourseContainerViewModel: BaseCourseViewModel {
         do {
             if isInternetAvaliable {
                 courseStructure = try await interactor.getCourseBlocks(courseID: courseID)
+                Task {
+                    await courseDatesViewModel.getCourseDates(courseID: courseID)
+                }
```

Comment:
@shafqat-muneer, I don't know if this can be tested. We can't check what's going on inside the Task.
And this dependency on CourseDatesViewViewModel 🤔.
But I can't offer a good solution right now. Only we could send notification to CourseDatesViewModel instead of calling it directly. 
What do you think?

In any case, the calendar flow will be heavily modified in FC-0047, so we can probably make do with this approach for now.